### PR TITLE
Added `render_params` argument in `render_model`

### PR DESCRIPTION
**Pull request by karm-patel**
Hi @fehiepsi and team, regarding issue #1379, I've made the following changes,

1. Added `render_params` argument in `render_model`
2. Added keys `sample_param` and `param_constraint` in dictionary returned by `get_model_relations()`
3. Enabled Provenance tracking for params in `get_model_relations` to add values in `sample_param`. 
4. Added example regarding the use of `render_params` in the tutorial `model_rendering.ipynb`

@fehiepsi, I would appreciate your code review!

### CODE REVIEW

File: notebooks/source/model_rendering.ipynb [hunk 1]
```diff
@@ -448,32 +448,39 @@
...
+   "id": "645df936",
    "metadata": {},
    "outputs": [],
    "source": [
     "def model(data):\n",
-    "    x = numpyro.sample(\"x\", dist.Normal(0, 1))\n",
-    "    y = numpyro.sample(\"y\", dist.LogNormal(x, 1))\n",
+    "    m = numpyro.param(\"m\", jnp.array(0))\n",
+    "    sd = numpyro.param(\"sd\", jnp.array([1]), constraint=constraints.positive)\n",
+    "    lambd = numpyro.sample(\"lambda\", dist.Normal(m, sd))\n",
```
Comment (fehiepsi) [inline notebooks/source/model_rendering.ipynb:475]: I think we can use `dist.LogNormal` to make sure `lambd` is positive.

File: notebooks/source/model_rendering.ipynb [hunk 2]
```diff
@@ -448,32 +448,39 @@
...
+   "execution_count": 26,
+   "id": "645df936",
    "metadata": {},
    "outputs": [],
    "source": [
     "def model(data):\n",
-    "    x = numpyro.sample(\"x\", dist.Normal(0, 1))\n",
-    "    y = numpyro.sample(\"y\", dist.LogNormal(x, 1))\n",
+    "    m = numpyro.param(\"m\", jnp.array(0))\n",
+    "    sd = numpyro.param(\"sd\", jnp.array([1]), constraint=constraints.positive)\n",
```
[the remark this answers is not in the document]
Comment (karm-patel) [reply notebooks/source/model_rendering.ipynb:476]: Okay, done.

File: numpyro/contrib/render.py [hunk 3]
```diff
@@ -129,26 +134,53 @@ def get_log_probs(sample):
     samples = {
         name: ProvenanceArray(site["value"], frozenset({name}))
         for name, site in trace.items()
-        if site["type"] == "sample" and not site["is_observed"]
+        if (site["type"] == "sample" and not site["is_observed"])
+    }
+
+    params = {
+        name: ProvenanceArray(site["value"], frozenset({name}))
```
[the remark this answers is not in the document]
Comment (karm-patel) [reply numpyro/contrib/render.py]: Interesting!, Thanks @fehiepsi  for pointing out this issue. Probably I have fixed that. Please check following output
```python
def model(data):
    lambda_base = numpyro.sample("lambda", dist.Normal(0, 1))
    net = flax_module("affine_net", flax_nn.Dense(1), input_shape=(1,))
    lambd = jnp.exp(net(jnp.expand_dims(lambda_base, -1)).squeeze(-1))
    with numpyro.plate("N", len(data)):
        numpyro.sample("obs", dist.Exponential(lambd), obs=data)

numpyro.render_model(model,model_args=(data,),render_distributions=True, render_params=True)
```
image


But I'm not sure where to add this example, so I have added this example in `model_rendering.ipynb` with the title "Rendering neural network's parameters". Feel free to suggest any edits.

File: notebooks/source/model_rendering.ipynb (hunk 1 above)
Comment (karm-patel) [reply notebooks/source/model_rendering.ipynb]: Ohh thanks, I corrected it.

File: numpyro/contrib/render.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (fehiepsi) [reply numpyro/contrib/render.py]: Whoa, it is actually working. Thanks, @karm-patel! Do you think it is better to strip out the`$params` part in `render_model` (at `cur_graph.node(rv, label=rv, shape=shape, style="filled", fillcolor=color)` line)?

File: notebooks/source/model_rendering.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (fehiepsi) [reply notebooks/source/model_rendering.ipynb]: Oops, sorry, I think you can use `jnp.result_type(data)` here.

**Review by fehiepsi — APPROVED** [review]
Looks great to me! This is super useful, especially when we use neural networks. We can see clear that which networks are used for some random variables. There are two nits as commented above.

File: notebooks/source/model_rendering.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (karm-patel) [reply notebooks/source/model_rendering.ipynb]: cool! I changed it.

File: numpyro/contrib/render.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (karm-patel) [reply numpyro/contrib/render.py]: Yes, I removed it, @fehiepsi  please check it.

File: notebooks/source/model_rendering.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (fehiepsi) [reply notebooks/source/model_rendering.ipynb]: Could you also update the model in the notebook (jnp.array(0.) -> 0.) and rerun it for the new $params removed.

[the remark this answers is not in the document]
Comment (karm-patel) [reply notebooks/source/model_rendering.ipynb]: Sure, I updated it and re-ran the notebook.

**Review by fehiepsi — APPROVED** [review]
